SensorFusion/Src/SensorFusion.cpp

```diff
@@ -590,55 +580,54 @@ SFError_t SF_GenerateNewResult()
         &imuData, &attitudeOutput, &iterData);
     if(SFError.errorCode != 0) return SFError;
 #else
-    pathOutput.altitude = 0;
-    pathOutput.rateOfClimb = 0;
-    pathOutput.latitude = 0;
-    pathOutput.latitudeSpeed = 0;
-    pathOutput.longitude = 0;
-    pathOutput.longitudeSpeed = 0;
-    pathOutput.track = 0;
-    pathOutput.groundSpeed = 0;
+	pathOutput.altitude = 0;
+	pathOutput.rateOfClimb = 0;
+	pathOutput.latitude = 0;
+	pathOutput.latitudeSpeed = 0;
+	pathOutput.longitude = 0;
+	pathOutput.longitudeSpeed = 0;
+	pathOutput.track = 0;
+	pathOutput.groundSpeed = 0;
 #endif
 
-    SFOutput.pitch = attitudeOutput.pitch;
-    SFOutput.roll = attitudeOutput.roll;
-    SFOutput.yaw = attitudeOutput.yaw;
-    SFOutput.pitchRate = attitudeOutput.pitchRate;
-    SFOutput.rollRate = attitudeOutput.rollRate;
-    SFOutput.yawRate = attitudeOutput.yawRate;
-    SFOutput.altitude = pathOutput.altitude;
-    SFOutput.rateOfClimb = pathOutput.rateOfClimb;
-    SFOutput.latitude = pathOutput.latitude;
-    SFOutput.latitudeSpeed = pathOutput.latitudeSpeed;
-    SFOutput.longitude = pathOutput.longitude;
-    SFOutput.longitudeSpeed = pathOutput.longitudeSpeed;
-    SFOutput.track = pathOutput.track;
-    SFOutput.groundSpeed = pathOutput.groundSpeed;
-    SFOutput.heading = attitudeOutput.heading;
-
-    return SFError;
+	SFOutput.pitch = attitudeOutput.pitch;
+	SFOutput.roll = attitudeOutput.roll;
+	SFOutput.yaw = attitudeOutput.yaw;
+	SFOutput.pitchRate = attitudeOutput.pitchRate;
+	SFOutput.rollRate = attitudeOutput.rollRate;
+	SFOutput.yawRate = attitudeOutput.yawRate;
+	SFOutput.altitude = pathOutput.altitude;
+	SFOutput.rateOfClimb = pathOutput.rateOfClimb;
+	SFOutput.latitude = pathOutput.latitude;
+	SFOutput.latitudeSpeed = pathOutput.latitudeSpeed;
+	SFOutput.longitude = pathOutput.longitude;
+	SFOutput.longitudeSpeed = pathOutput.longitudeSpeed;
+	SFOutput.track = pathOutput.track;
+	SFOutput.groundSpeed = pathOutput.groundSpeed;
+	SFOutput.heading = attitudeOutput.heading;
+
+	return SFError;
 }
 
-SFError_t SF_GetResult(SFOutput_t *output)
-{
-    SFError_t SFError;
-    SFError.errorCode = 0;
+SFError_t SF_GetResult(SFOutput_t *output) {
+	SFError_t SFError;
+	SFError.errorCode = 0;
 
-    *output = SFOutput;
+	*output = SFOutput;
 
-    return SFError;
+	return SFError;
 }
 
-IMU_Data_t SF_GetRawIMU()
-{
-    IMUData_t imuData;
-    imuObj->GetResult(imuData);
-
-    IMU_Data_t imuOutput;
+IMU_Data_t SF_GetRawIMU() {
+	IMUData_t imuData;
+	//imuObj->GetResult(imuData);
+	IMU& imuObj = BMX160::getInstance();
```

Comment:
I've noticed that the IMU has been declared a few times in local scope. Is there a reason it is no longer a global object? Is the allocation time to create/destroy these worth the ease of reading?
I can't remember if  the getInstance just returns a pointer to the same existing object, in which case this is probaby not an issue